### PR TITLE
[Term Entry] Go Math Functions .Modf() 

### PR DESCRIPTION
**Pull request by MiraVyslovukh**
### Description

- Added the Modf() ferm for Go

### Type of Change

- Adding a new entry
- Closes issue #2830 

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/go/concepts/math-functions/terms/modf/modf.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  - 'learn-go'
+  - 'paths/computer-science'
+---
+
+The **`Modf()`** function in Go is used to find the integer and fractional part of a floating-point number. This function is part of the `math` package, and the package must be imported to use the function in a Go program.
+
+## Syntax
+
+```pseudo
+result := Modf(f float64) (int float64, frac float64)
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/modf/modf.md:21]:
```suggestion
result := Modf(f) (int, frac)
```

File: content/go/concepts/math-functions/terms/modf/modf.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+Where:
+
+- `f`: is the floating-point number for which you want to extract the integer and fractional parts.
+- `float64`: is a type of numer that uses 64 bytes.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/modf/modf.md:27]:
```suggestion
```

File: content/go/concepts/math-functions/terms/modf/modf.md (hunk 1 above)
Comment (caupolicandiaz) [reply content/go/concepts/math-functions/terms/modf/modf.md:21]: Typically in docs we keep the syntax expression very minimal, and highlight types and other details in the section below that describes any parameters or special cases, etc

File: content/go/concepts/math-functions/terms/modf/modf.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+- `int`: is the integer part of the passed parameter.
+- `frac`: is the fractional part of the passed parameter.
+
+Some special return cases:
+
+- If (±)Inf is passed as an argument, then `Modf` returns (±)Inf and `NAN`.
+- If the argument passed is `NAN`, then Modf returns `NAN` and `NAN`.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/modf/modf.md:34]:
```suggestion
- If the argument passed is `NAN`, then the return will also be `NAN`.
```

File: content/go/concepts/math-functions/terms/modf/modf.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+## Example
+
+Below is a simple example where we find out the integer and fractional parts of 3.3456.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/modf/modf.md:38]:
```suggestion
Below is a basic implementation of the `Modf()` function.
```

File: content/go/concepts/math-functions/terms/modf/modf.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+import(
+  "fmt"
+  "math"
+)
+
+func main() {
+  var x float64 = 3.3456
+
+  var w, z float64 = math.Modf(x)
+  fmt.Print("For ", x, " integer part: ", w, " fractional part: ", z)
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/modf/modf.md:52]:
```suggestion
  fmt.Print("The value ", x, " is composed of the integer: ", w, " and the fractional part: ", z)
```

File: content/go/concepts/math-functions/terms/modf/modf.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+}
+```
+
+The output will be:
+
+```shell
+For 3.3456 integer part: 3 fractional part: 0.34560000000000013
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/modf/modf.md:59]: update as edited above

File: content/go/concepts/math-functions/terms/modf/modf.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+```codebyte/golang
+package main
+
+import (
+    "fmt"
+    "math"
+)
+
+func main() {
+    // Contain both integer and fraction
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/modf/modf.md:75]:
```suggestion
    // A value with an integer and fraction
```

File: content/go/concepts/math-functions/terms/modf/modf.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    integer, fraction := math.Modf(2.5)
+    fmt.Printf("Integer: %f. Fraction: %f\n", integer, fraction)
+
+    // Contain only integer part
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/modf/modf.md:79]:
```suggestion
    // An integer
```

File: content/go/concepts/math-functions/terms/modf/modf.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    integer, fraction = math.Modf(2)
+    fmt.Printf("Integer: %f. Fraction: %f\n", integer, fraction)
+
+    // Negative floating point number
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/modf/modf.md:83]:
```suggestion
    // A negative floating point number
```

File: content/go/concepts/math-functions/terms/modf/modf.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+    integer, fraction = math.Modf(-2.5)
+    fmt.Printf("Integer: %f. Fraction: %f\n", integer, fraction)
+
+    // Contains only fraction part
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/modf/modf.md:87]:
```suggestion
    // A number of a fractional value
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@MiraVyslovukh I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

File: content/go/concepts/math-functions/terms/modf/modf.md (hunk 2 above)
Comment (caupolicandiaz) [reply content/go/concepts/math-functions/terms/modf/modf.md:27]: lines 28 and 29 should be edited so that they are presented as returns not as parameters

Comment (MiraVyslovukh) [reply content/go/concepts/math-functions/terms/modf/modf.md]: Hey @caupolicandiaz, does it sound better if I update it to this?
- `int`: is a returned integer part value.
- `frac`: is a returned fractional part value.

**Review by caupolicandiaz — APPROVED** [review]
@MiraVyslovukh I pushed one minor edit regarding the syntax. This is ready for a second review.

File: content/go/concepts/math-functions/terms/modf/modf.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+
+## Syntax
+
+```pseudo
+int, frac := Modf(f)
+```
+
+Where:
+
+- `f`: is the floating-point number for which you want to extract the integer and fractional parts.
```
Comment (SSwiniarski) [inline content/go/concepts/math-functions/terms/modf/modf.md:26]:
```suggestion
- `f`: is the floating-point number to extract the integer and fractional parts from.
```

File: content/go/concepts/math-functions/terms/modf/modf.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+
+  var w, z float64 = math.Modf(x)
+  fmt.Print("The value ", x, " is composed of the integer: ", w, " and the fractional part: ", z)
+}
+```
+
+The output will be:
+
+```shell
+For 3.3456 integer part: 3 fractional part: 0.34560000000000013
```
Comment (SSwiniarski) [inline content/go/concepts/math-functions/terms/modf/modf.md:58]:
```suggestion
The value 3.3456 is composed of the integer: 3 and the fractional part: 0.34560000000000013
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@MiraVyslovukh I did a second review and had a couple of comments.

**Review by SSwiniarski — APPROVED** [review]
@MiraVyslovukh Looks good.

**Comment by SSwiniarski** [comment]
@MiraVyslovukh Congrats on the new entry! Live link here: https://www.codecademy.com/resources/docs/go/math-functions/modf

image

You're currently anonymous. If you don't wish to be anonymous, check out this forum thread on how to link your GitHub and Codecademy profiles: https://discuss.codecademy.com/t/my-contribution-is-live-but-im-not-being-shown-as-a-contributor-why-is-this/758036  **Note:** it may take a few minutes for the change to register on the entry page.

**Pull request merged**